hyko_toolkit/functions/web/web_scraping/scrapy_crawler/metadata.py

```diff
@@ -0,0 +1,35 @@
+from hyko_sdk.models import CoreModel
+from pydantic import Field
+
+from hyko_toolkit.registry import ToolkitFunction
+
+func = ToolkitFunction(
+    name="scrapy_crawler",
+    task="web_scraping",
+    description="use scrapy to crawl web pages",
+    absolute_dockerfile_path="./toolkit/hyko_toolkit/functions/web/web_scraping/scrapy_crawler/Dockerfile",
+    docker_context="./toolkit/hyko_toolkit/functions/web/web_scraping/scrapy_crawler",
+)
+
+
+@func.set_input
+class Inputs(CoreModel):
+    start_urls: str = Field(
+        ...,
+        description="A starting url to crawl. Protocol must be either 'http' or 'https'.",
+    )
+    stop_urls: str = Field(
+        ...,
+        description="A list of urls to stop crawling. Protocol must be either 'http' or 'https'.",
+    )
+
+
+@func.set_param
+class Params(CoreModel):
+    allowed_domains: str = Field(..., description="A domain to crawl.")
```

Comment:
is this one domain or a list of domains ? can this be optional and instead have depth of search ?

Reply:
Passing a single domain or a list of domains is mandatory; making it optional would result in errors.